test/e2e/io_soak/io_soak_test.go

```diff
@@ -0,0 +1,229 @@
+	nodeList, err := common.GetNodeLocs()
+	Expect(err).ToNot(HaveOccurred())
+
+	numMayastorNodes := 0
+	jobCount := 0
+	sort.Slice(nodeList, func(i, j int) bool { return nodeList[i].NodeName < nodeList[j].NodeName })
+	for i, node := range nodeList {
+		if node.MayastorNode && !node.MasterNode {
+			logf.Log.Info("MayastorNode", "name", node.NodeName, "index", i)
+			jobCount += loadFactor
+			numMayastorNodes += 1
+		}
+	}
+
+	Expect(replicas <= numMayastorNodes).To(BeTrue())
+	logf.Log.Info("IOSoakTest", "jobs", jobCount, "volumes", jobCount, "test pods", jobCount)
+
+	for _, proto := range protocols {
+		scName := fmt.Sprintf("io-soak-%s", proto)
+		logf.Log.Info("Creating", "storage class", scName)
+		err = common.MkStorageClass(scName, replicas, proto, "io.openebs.csi-mayastor")
+		Expect(err).ToNot(HaveOccurred())
+		scNames = append(scNames, scName)
+	}
+
+	// Create the set of jobs
+	idx := 1
+	for idx <= jobCount {
+		for _, scName := range scNames {
+			if idx > jobCount { break }
+			logf.Log.Info("Creating", "job", "fio filesystem job", "id", idx)
+			jobs = append(jobs, MakeFioFsJob(scName, idx))
+			idx++
+
+			if idx > jobCount { break }
+			logf.Log.Info("Creating", "job", "fio raw block job", "id", idx)
+			jobs = append(jobs, MakeFioRawBlockJob(scName, idx))
+			idx++
+		}
+	}
+
+	logf.Log.Info("Creating volumes")
+	// Create the job volumes
+	for _, job := range jobs {
+		job.makeVolume()
+	}
+
+	logf.Log.Info("Creating test pods")
+	// Create the job test pods
+	for _, job := range jobs {
+		pod, err := job.makeTestPod()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(pod).ToNot(BeNil())
+	}
+
+	logf.Log.Info("Waiting for test pods to be ready")
+	// Wait for the test pods to be ready
+	for _, job := range jobs {
+		// Wait for the test Pod to transition to running
+		Eventually(func() bool {
+			return common.IsPodRunning(job.getPodName())
+		},
+			defTimeoutSecs,
+			"1s",
+		).Should(Equal(true))
+	}
+
+	logf.Log.Info("Starting test execution in all test pods")
+	// Run the test jobs
+	doneC, errC := make(chan string), make(chan error)
+	go monitor(errC)
+	for _, job := range jobs {
+		go job.run(duration, doneC, errC)
+	}
+
+	logf.Log.Info("Waiting for test execution to complete on all test pods")
+	// Wait and check that all test pods have executed successfully
+	for range jobs {
+		select {
+		case podName := <-doneC:
+			logf.Log.Info("Completed", "pod", podName)
+		case err := <-errC:
+			close(doneC)
+			logf.Log.Error(err, "fio run")
+			Expect(err).To(BeNil())
+		}
+	}
+
+	logf.Log.Info("All runs complete, deleting test pods")
+	for _, job := range jobs {
+		err := job.removeTestPod()
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	logf.Log.Info("All runs complete, deleting volumes")
+	for _, job := range jobs {
+		job.removeVolume()
+	}
+
+	logf.Log.Info("All runs complete, deleting storage classes")
+	for _, scName := range scNames {
+		err = common.RmStorageClass(scName)
+		Expect(err).ToNot(HaveOccurred())
+	}
```

Comment:
I can see in the `ÀfterEach` section that we remove the storage class. Does the uninstall remove the volumes and test pods?

Reply:
Ah, good spot! the cleanup in AfterEach is a leftover from when I was debugging. Removed from AfterEach otherwise postmortem analysis will be "hampered"